### PR TITLE
[ENT-644] Remove migrations of old consent for a safe deployment. (Step 1)

### PR DESCRIPTION
**Pull request by UmanShahzad**
**Description:** This removes the migration involved in removal of the old account-level consent feature, so that we can deploy the code safely.

**JIRA:** [ENT-644](https://openedx.atlassian.net/browse/ENT-644)

**Merge deadline:** asap

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @UmanShahzad! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-enterprise&number=196"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by UmanShahzad** [comment]
@douglashall Not sure why the tests are failing. Is this the proper way of going about this?

File: enterprise/migrations/0026_remove_old_consent.py [hunk 1]
```diff
@@ -1,48 +0,0 @@
...
-            name='user',
-        ),
-        migrations.RemoveField(
-            model_name='enterprisecourseenrollment',
-            name='consent_granted',
-        ),
-        migrations.RemoveField(
-            model_name='enterprisecustomer',
-            name='require_account_level_consent',
-        ),
```
Comment (douglashall) [inline enterprise/migrations/0026_remove_old_consent.py]: I think before we entirely remove this field from code, we must first mark it as nullable and deploy the migration associated with that change. We can then remove the field entirely from code and deploy, then deploy the migration to remove it from the database.

Comment (UmanShahzad) [reply enterprise/migrations/0026_remove_old_consent.py]: That sounds pretty complex, and I'm starting to wonder if this is necessary for work outside of the platform (I wouldn't imagine we could *ever* do this for external dependencies that made such a migration, for example, but things work out fine). The main reason I'm doubtful is that, in order for this migration to come up *in the first place*, we have to have installed the latest edx-enterprise version, which comes with the latest code. I could still be wrong so maybe we should double-check with ops?

Comment (douglashall) [reply enterprise/migrations/0026_remove_old_consent.py]: It is complex, but it is necessary. I imagine that if we have done something similar in the past that the feature which touched the models with removed fields was broken for the time between when we run the migrations and when the servers with the new code start servicing users.

Comment (UmanShahzad) [reply enterprise/migrations/0026_remove_old_consent.py]: Alright, so I'll add back the fields in this PR and make them nullable, and generate the migration for that and we can deploy it. In #197 I'll remove these fields only but not make any new migration. In a new PR we'll finally make the migration. Jeez :P

**Comment by UmanShahzad** [comment]
@douglashall Alrighty, can you double-check that this ok as "step 1" in this process?

**Comment by UmanShahzad** [comment]
@haikuginger Can I get your quick +1 on this?

**Comment by haikuginger** [comment]
@douglashall, do we need to be similarly squeamish about the consent field that exists on the EnterpriseCourseEnrollment?

**Comment by douglashall** [comment]
@haikuginger The `EnterpriseCourseEnrollment.consent` field is currently nullable, so we should be good there. We will need to deploy the removal from code of that field first, before deploying the migration that removes it from the database.

**Comment by UmanShahzad** [comment]
@douglashall @haikuginger Just a note on the priority of this: this safe deployment is blocking the edx-enterprise version bump pipeline we've got going on over at edx-platform. We can't bring any more changes here onto stage and prod until we safely deploy this because 0.46.5 contains the dangerous migration.

**Comment by UmanShahzad** [comment]
@haikuginger Ah yeah, sorry if that was confusing, we've been going back and forth with this but it's clear now what the path is. Thanks a lot!

**Pull request merged**